src/Symfony/Component/Console/Input/InputOption.php

```diff
@@ -259,4 +294,14 @@ public function equals(self $option): bool
             && $option->isValueOptional() === $this->isValueOptional()
         ;
     }
+
+    /**
+     * Returns true if the option allows $mode.
+     *
+     * @return bool true if mode is $mode, false otherwise
+     */
+    protected function hasMode(int $mode): bool
```

Comment:
I'd rather not add a new method, it's unrelated to the proposed feature

Reply:
Simplifies and reduces duplicate/similar code

Reply:
This method is not added as an internal method. You added it as part of the semver API.

If you want to keep the helper, make it private instead.